cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/resource/AzureVolumeResourceBuilder.java

```diff
@@ -224,13 +229,17 @@ public CloudResource delete(AzureContext context, AuthenticatedContext auth, Clo
             List<String> volumeIds = getVolumeSetAttributes(cloudResource).getVolumes().stream()
                     .map(VolumeSetAttributes.Volume::getId)
                     .collect(Collectors.toList());
+            LOGGER.info("VolumeIds to delete: {}", volumeIds);
             for (String volumeId : volumeIds) {
-                if (ResourceStatus.ATTACHED.equals(cloudResourceStatus.getStatus())) {
-                    CloudResource resourceGroup = context.getNetworkResources().stream()
-                            .filter(r -> r.getType().equals(ResourceType.AZURE_RESOURCE_GROUP))
-                            .findFirst()
-                            .orElseThrow(() -> new AzureResourceException("Resource group resource not found"));
+                CloudResource resourceGroup = context.getNetworkResources().stream()
+                        .filter(r -> r.getType().equals(ResourceType.AZURE_RESOURCE_GROUP))
+                        .findFirst()
+                        .orElseThrow(() -> new AzureResourceException("Resource group resource not found"));
+                try {
+                    LOGGER.info("Trying to detach volume {} from {}", volumeId, instanceName);
                     client.detachDiskFromVm(volumeId, client.getVirtualMachine(resourceGroup.getName(), instanceName));
+                } catch (RuntimeException e) {
+                    LOGGER.info("Can not detach {} from {}", volumeId, instanceName, e);
```

Comment:
Logger:
public void debug(String msg, Throwable t);

you need to String.format because `e` would be swallowed (but would be printed only as a toString() if it had a placeholder)